### PR TITLE
Convert ansible to dockerfile

### PR DESCRIPTION
**Pull request by mumarkhan999**
**ISSUE:** https://github.com/openedx/devstack/issues/962

This PR is part of effort aimed at removing Ansible based configurations and replacing them with Dockerfile. Currently Devstack Docker images are built using Ansible based configurations in the configurations repository. Through this effort we will make sure that the Repo has its own Dockerfile which has all the necessary configurations to setup small production and dev environments.

**Steps to run this Image with Devstack:**

- Build the Image locally first using the target `dev` i.e. `docker build -t image-name-of-choice --target dev .`
- After the image is built successfully go to the `docker compose` file of `devstack` and replace the existing `registrar` image with the one that you built without changing any other configurations there.
- Run `make dev.up.credentials` in the terminal while in the devstack directory.

### CODE REVIEW

File: scripts/devstack.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Ansible managed
```
Comment (aht007) [inline scripts/devstack.sh:3]: nit: this line should be removed

File: Dockerfile [hunk 2]
```diff
@@ -1,112 +1,140 @@
...
+COPY scripts/credentials.sh $CREDENTIALS_APP_DIR/credentials.sh
+COPY /configuration_files/credentials.conf ${SUPERVISOR_AVAILABLE_DIR}/credentials.conf
+COPY /configuration_files/credentials.conf ${SUPERVISOR_CFG_DIR}/credentials.conf
+# Manage.py symlink
+COPY /manage.py /edx/bin/manage.credentials
+
+RUN chown credentials:credentials "$CREDENTIALS_APP_DIR/devstack.sh" && chmod a+x "$CREDENTIALS_APP_DIR/devstack.sh"
+
+# placeholder file for the time being unless devstack provisioning scripts need it.
+RUN touch ${CREDENTIALS_APP_DIR}/credentials_env
```
Comment (aht007) [inline Dockerfile:115]: This line should be moved to the dev target

File: Dockerfile [hunk 3]
```diff
@@ -1,112 +1,140 @@
...
-RUN touch ../credentials_env
 
-CMD while true; do python ./manage.py runserver 0.0.0.0:18150; sleep 2; done
+FROM app as production
+
+ENV DJANGO_SETTINGS_MODULE credentials.settings.production
+
+COPY scripts/credentials.sh "$CREDENTIALS_APP_DIR/credentials.sh"
+
+#CMD ["gunicorn", "--workers=2", "--name", "credentials", "-c", "/edx/app/credentials/credentials/credentials/docker_gunicorn_configuration.py", "--log-file", "-", "--max-requests=1000", "credentials.wsgi:application"]
```
Comment (aht007) [inline Dockerfile:126]: nit: Remove the old comments from the file

File: Dockerfile [hunk 4]
```diff
@@ -1,112 +1,140 @@
...
-apt-get install -y software-properties-common && \
-apt-add-repository -y ppa:deadsnakes/ppa && apt-get update && \
-apt-get upgrade -qy && apt-get install language-pack-en locales git \
-python3.8-dev python3.8-venv libmysqlclient-dev libssl-dev build-essential wget unzip -qy && \
-rm -rf /var/lib/apt/lists/*
+RUN addgroup credentials
+RUN adduser --disabled-login --disabled-password credentials --ingroup credentials
+
+
+RUN mkdir -p "$CREDENTIALS_APP_DIR"
```
Comment (aht007) [inline Dockerfile:53]: We don't need to explicitly call the `mkdir` command here as specifying `WORKDIR ${CREDENTIALS_CODE_DIR}` would make all the directories that precede the code dir path

**Comment by hurtstotouchfire** [comment]
FYI, we will take a look a this next week.

**Comment by Tj-Tracy** [comment]
LGTM, I'll merge 👍

**Review by Tj-Tracy — APPROVED** [review]

**Comment by mumarkhan999** [comment]
Hi @hurtstotouchfire @Tj-Tracy ,

This epic related to this PR is still in discovery work and we are getting a lot of input from SRE and Arch side, hence this wasn’t supposed to be merged right now. According to the input we received we need to do some changes:

- Move configuration files from this repo to devstack
- Remove supervisor related stuff from dockerfile
- Update CI/CD flow of pushing docker image

This PR failed on the master branch as we didn’t have a build target in our new dockerfile but CI workflow was trying to push the build target.

Now I've created a new PR https://github.com/openedx/credentials/pull/1782
Once the above-mentioned things have been done, we'll let you know to review it.

**Comment by Tj-Tracy** [comment]
@mumarkhan999 appreciate the followup 👍 sorry for the early merge.

**Pull request merged**